src/c-writer.cc

```diff
@@ -2712,6 +2943,110 @@ void CWriter::Write(const Func& func) {
   func_ = nullptr;
 }
 
+void CWriter::WriteTailCallee(const Func& func) {
+  func_ = &func;
+  in_tail_callee_ = true;
+  local_syms_.clear();
+  local_sym_map_.clear();
+  stack_var_sym_map_.clear();
+  func_sections_.clear();
+  func_includes_.clear();
+
+  Stream* prev_stream = stream_;
```

Comment:
Is it worth extracting these 10 lines into a new helper method?  e.g. `BeginFunction`?

Reply:
Done.